### PR TITLE
Refactoring of ConvertUnary

### PR DESCRIPTION
**Pull request by drivanov**
ConvertUnary is rewritten using OpConverterBase

### CODE REVIEW

File: tensorflow/compiler/tf2tensorrt/convert/ops/unary_ops.cc
Comment (christopherbate) [inline tensorflow/compiler/tf2tensorrt/convert/ops/unary_ops.cc:46]: This is checked in the base class via the InputSpec() you defiend above. So you shouldn't need this function call and also not need to declare `CheckInputsWeightrs` in the header

Comment (drivanov) [reply tensorflow/compiler/tf2tensorrt/convert/ops/unary_ops.cc]: Good point! Fixed.

**Review by tfeher — APPROVED** [review]
Thanks @drivanov for refactoring the converter! It looks good to me.

**Comment by bixia1** [comment]
@drivanov Would you please fix the first conversation block. When creating new PRs:
First conversation block: put only the information that you want to go with the commit message there.
Second conversation block: put more details, such as related PRs, background of the work, explanation of the problem trying to solve and the approach taken, etc.

**Comment by bixia1** [comment]
I will approve this after the other two PRs are merged.

**Comment by drivanov** [comment]
NOTE: This PR should be slightly adjusted if it will be merged after  
[PR#54181](https://github.com/tensorflow/tensorflow/pull/54181):**Implementation of converters for unary operators "Round" and "Sign"**
[PR#53718](https://github.com/tensorflow/tensorflow/pull/53718): **Converter for Tile operation**

**Comment by bixia1** [comment]
@drivanov would you please rebase and squash the commits into one?

**Comment by drivanov** [comment]
[PR#55082](https://github.com/tensorflow/tensorflow/pull/55082) is a substitute for that one.

**Comment by drivanov** [comment]
[PR#55082](https://github.com/tensorflow/tensorflow/pull/55082)  is a substitute for that one.

**Pull request closed without merging**
